x-pack/plugins/snapshot_restore/public/app/components/repository_delete_provider.tsx

```diff
@@ -4,26 +4,177 @@
+
 export const RepositoryDeleteProvider: React.FunctionComponent<Props> = ({ children }) => {
+  const {
+    core: {
+      i18n,
+      notification: { toastNotifications },
+    },
+  } = useAppDependencies();
+  const { FormattedMessage } = i18n;
   const [repositoryNames, setRepositoryNames] = useState<Array<Repository['name']>>([]);
   const [isModalOpen, setIsModalOpen] = useState<boolean>(false);
-  const deleteRepository: DeleteRepository = names => {
+  const onSuccessCallback = useRef<OnSuccessCallback | null>(null);
+
+  const confirmDeleteRepository: DeleteRepository = (
+    names: Array<Repository['name']>,
+    onSuccess?: OnSuccessCallback
+  ): void => {
     setIsModalOpen(true);
     setRepositoryNames(names);
+    if (typeof onSuccess === 'function') {
+      onSuccessCallback.current = onSuccess;
+    } else {
+      onSuccessCallback.current = null;
+    }
   };
 
-  if (isModalOpen && repositoryNames.length) {
-    /* placeholder */
-  }
+  const closeModal = () => {
+    setIsModalOpen(false);
+    setRepositoryNames([]);
+  };
+
+  const deleteRepository = () => {
+    const repositoriesToDelete = [...repositoryNames];
+    deleteRepositories(repositoriesToDelete).then(({ data, error }) => {
+      // Surface success notifications
+      if (data.success && data.success.length) {
+        const hasMultipleSuccesses = data.success.length > 1;
+        const successMessage = hasMultipleSuccesses
+          ? i18n.translate(
+              'xpack.snapshotRestore.deleteRepository.successMultipleNotificationTitle',
+              {
+                defaultMessage: 'Removed {count} repositories',
+                values: { count: data.success.length },
+              }
+            )
+          : i18n.translate(
+              'xpack.snapshotRestore.deleteRepository.successSingleNotificationTitle',
+              {
+                defaultMessage: "Removed repository '{name}'",
+                values: { name: data.success[0] },
+              }
+            );
+        toastNotifications.addSuccess(successMessage);
+        if (onSuccessCallback.current) {
+          onSuccessCallback.current([...data.success]);
+        }
+      }
+
+      // Surface error notifications
+      // `error` is generic server error
+      // `data.error` are specific errors with removing particular repository(ies)
+      if (error || (data.error && data.error.length)) {
+        const hasMultipleErrors =
+          (data.error && data.error.length > 1) || (error && repositoriesToDelete.length > 1);
+        const errorMessage = hasMultipleErrors
+          ? i18n.translate(
+              'xpack.snapshotRestore.deleteRepository.errorMultipleNotificationTitle',
+              {
+                defaultMessage: 'Error removing {count} repositories',
+                values: {
+                  count: (data.errors && data.errors.length) || repositoriesToDelete.length,
```

Comment:
It looks like `data.errors` is a typo here and on line 97, since we're expecting `data.error` right?